### PR TITLE
chore: update docs and ci

### PR DESCRIPTION
**Pull request by Fdawgs**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by zekth — APPROVED** [review]
lgtm

**Review by mcollina — CHANGES_REQUESTED** [review]
Good work, just a nit

File: auth.js [hunk 1]
```diff
@@ -28,14 +28,14 @@ function auth (functions, opts) {
     throw new Error('The value of options.run must be \'all\'')
   }
 
-  for (var i = 0; i < functions.length; i++) {
+  for (let i = 0; i < functions.length; i++) {
```
[the remark this answers is not in the document]
Comment (Fdawgs) [reply auth.js:31]: Thanks @mcollina, I can't find anything in the nodejs documentation or elsewhere regarding `let` being slower than `var` in loops, apart from this https://github.com/nodejs/node/pull/30380 (includes you!), which doesn't really clarify why it's slower?

[the remark this answers is not in the document]
Comment (jsumners) [reply auth.js:31]: It's just the implementation of the keyword `let` in this context.

Comment (jsumners) [inline auth.js:31]:
```suggestion
  /* eslint-disable-next-line no-var */
  for (var i = 0; i < functions.length; i++) {
```

**Review by jsumners — APPROVED** [review]
LGTM

**Pull request merged**